pilot/lib/pilot/build_manager.rb

```diff
@@ -159,12 +159,18 @@ def update_beta_app_meta(options, build)
     end
 
     def self.truncate_changelog(changelog)
-      max_changelog_length = 4000
-      if changelog && changelog.length > max_changelog_length
-        original_length = changelog.length
+      # Apple documentation on "What's New in this Version" field, the one used for changelog
+      # https://help.apple.com/app-store-connect/?lang=en#/devf29afbb74
+      max_changelog_bytesize = 4000
+      if changelog && changelog.bytesize > max_changelog_bytesize
+        original_bytesize = changelog.bytesize
         bottom_message = "..."
-        changelog = "#{changelog[0...max_changelog_length - bottom_message.length]}#{bottom_message}"
-        UI.important("Changelog has been truncated since it exceeds Apple's #{max_changelog_length} character limit. It currently contains #{original_length} characters.")
+        truncated_bytes = changelog.bytes[0...max_changelog_bytesize - bottom_message.length]
+        # We convert the bytes back to a string, which can contained invalid bytes that
+        # do not form a character
+        unscrubbed_truncated_string = truncated_bytes.pack("C*").force_encoding("UTF-8")
+        changelog = "#{unscrubbed_truncated_string.scrub('')}#{bottom_message}"
+        UI.important("Changelog has been truncated since it exceeds Apple's #{max_changelog_bytesize} character limit. It currently contains #{original_bytesize} characters.")
```

Comment:
Should we change `character limit` and `characters` to `byte limit` and `bytes?
```suggestion
        UI.important("Changelog has been truncated since it exceeds Apple's #{max_changelog_bytesize} byte limit. It currently contains #{original_bytesize} bytes.")
```

Reply:
Hi, My first commit before actually finding the documentation had changed character to byte because I felt it was more correct, given that I considered a kanji to be a character. However after reading Apple docs and investigating Unicode I saw the correct term is actually "multibyte character" so I put it back.

I believe your comment is about my original commit, not the latest.

Reply:
I was actually talking about the message that gets displayed to the user on your latest commit. If someone has 4000 kanji and it gets truncated someone might be confused by since its actually checking against 4000 bytes and not 4000 kanji characters. But the odds of that happening are probably pretty low so I'm okay with this 😊

Reply:
Ah sorry, I had confused the suggested change with the change I had already done.